django/thunderstore/community/factories.py

```diff
@@ -27,3 +29,71 @@ class Meta:
 
     community = factory.SubFactory(CommunityFactory)
     site = factory.SubFactory(SiteFactory)
+
+
+class PackageListingFactory(DjangoModelFactory):
+    class Meta:
+        model = PackageListing
+
+    class Params:
+        community_ = None
+        community_kwargs = {}
+        package_ = None
+        package_kwargs = {}
+        package_version_kwargs = {}
+
+    @factory.lazy_attribute
+    def community(self):
+        """
+        To use an existing community, pass it via community_ parameter.
+
+        To create a new package with custom values, pass the values via
+        community_kwargs parameter.
+        """
+        if self.community_:
+            return self.community_
+
+        site = CommunitySiteFactory()
+
+        if self.community_kwargs:
+            for attr, value in self.community_kwargs.items():
+                setattr(site.community, attr, value)
```

Comment:
For one thing, we could compare the fields in `community_kwargs` against `Community._meta.fields`.

But I guess in the long run it would be better to change the other factories to support this (i.e. first call `CommunityFactory` with the `community_kwargs`, and then call `CommunitySiteFactory` and pass the newly created `Community` to it.

Reply:
Agreed, but sounds like a change that should be done in a separate PR. Created an issue about it & closing this.